### PR TITLE
Removed inference from [TypeName] and underscore

### LINKED ISSUE #1303: Remove support for inferring [Type]Id properties.
Right now we infer Id's from either `Id` or `[TypeName]Id` properties, this is undocumented and does not feel very natural IMO. 

Lets remove support for inferring `[TypeName]Id` automattically in 2.0

### PR DESCRIPTION
**Pull request by gmoskovicz**
Closes #1303 

__Please review__

- [ ] @gmarz / @Mpdreamz

### CODE REVIEW

File: src/Nest/Resolvers/IdResolver.cs [hunk 1]
```diff
@@ -110,23 +110,11 @@ private PropertyInfo GetInferredId(Type type)
 			if (esTypeAtt != null && !string.IsNullOrWhiteSpace(esTypeAtt.IdProperty))
 				return GetPropertyCaseInsensitive(type, esTypeAtt.IdProperty);
 
-			propertyName = "Id";
-			//Try Id on its own case insensitive
-			var idProperty = GetPropertyCaseInsensitive(type, propertyName);
+			var idProperty = GetPropertyCaseInsensitive(type, "Id");
 			if (idProperty != null)
```
Comment (Mpdreamz) [inline src/Nest/Resolvers/IdResolver.cs]: use a ternary return here.

File: src/Tests/Nest.Tests.Unit/Internals/Inferno/IdLookupTests.cs [hunk 2]
```diff
@@ -100,6 +104,24 @@ public void TestInheritedLookup()
 		}
 
 		[Test]
+		public void TestTypedIdLookup()
+		{
+			var expected = new TypedIdClass() { TypedIdClassId = 123 };
+			var id = this._idResolver.GetIdFor(expected);
+			Assert.AreNotEqual(expected.TypedIdClassId, id);
```
Comment (Mpdreamz) [inline src/Tests/Nest.Tests.Unit/Internals/Inferno/IdLookupTests.cs]: Use FluentAssertations (`*.Should()`)for new tests over `Assert.*`

**Comment by Mpdreamz** [comment]
+1 for removing support for infirring `[TypeXName]Id` and `[TypeXName]_Id` properties on instance of `TypeX` in 2.0.

Very obscure and hardly used feature. @gmarz agree?

**Comment by Mpdreamz** [comment]
Also this PR has to be targeted against the 2.0 branch

**Comment by gmarz** [comment]
Yea, definitely agree.  I didn't even know this feature existed until you opened this issue. :)

**Comment by gmoskovicz** [comment]
Not longer valid.

**Pull request closed without merging**